pkg/oc/cli/admin/release/extract.go

```diff
@@ -118,15 +125,34 @@ func (o *ExtractOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args [
 }
 
 func (o *ExtractOptions) Run() error {
-	if len(o.From) == 0 {
-		return fmt.Errorf("must specify an image containing a release payload with --from")
+	sources := 0
+	if o.Tools {
+		sources++
 	}
-	if o.Directory != "." && len(o.File) > 0 {
-		return fmt.Errorf("only one of --to and --file may be set")
+	if len(o.File) > 0 {
+		sources++
+	}
+	if len(o.Command) > 0 {
+		sources++
 	}
-
 	if len(o.GitExtractDir) > 0 {
+		sources++
+	}
+
+	switch {
+	case sources > 1:
+		return fmt.Errorf("only one of --tools, --command, --file, or --git may be specified")
+	case len(o.From) == 0:
+		return fmt.Errorf("must specify an image containing a release payload with --from")
+	case o.Directory != "." && len(o.File) > 0:
+		return fmt.Errorf("only one of --to and --file may be set")
```

Comment:
I'd prefer we move those to `Validate` to follow the patterns we try to stick with in all `oc` commands.